code/obj/item/flamethrower.dm

```diff
@@ -588,7 +588,8 @@ ABSTRACT_TYPE(/obj/item/gun/flamethrower/backtank)
 	// PantsNote: Flamethrower disassmbly.
 	else if (isscrewingtool(W))
 		var/obj/item/gun/flamethrower/assembled/S = src
-		if (( S.gastank ))
+		if (( S.gastank || S.fueltank ))
+			boutput(user, SPAN_ALERT("You can't disassemble [src] while the tank is attached!"))
```

Comment:
slight verbiage nit since "the tank" is singular but it might have two tanks attached
```suggestion
			boutput(user, SPAN_ALERT("You can't disassemble [src] while it has an attached tank!"))
```